parlai/core/torch_generator_agent.py

```diff
@@ -735,12 +741,22 @@ def _generate(self, batch, beam_size, max_ts):
         if isinstance(model, torch.nn.parallel.DistributedDataParallel):
             model = self.model.module
         encoder_states = model.encoder(*self._model_input(batch))
-        dev = batch.text_vec.device
+        if batch.text_vec is not None:
+            dev = batch.text_vec.device
+        else:
```

Comment:
can you explain this?

Reply:
this is related to the long-standing issue that we still do not specify `batchsize` in a batch and instead have code that calls `batch.text_vec`; this breaks for any task that is not exclusively text based unfortunately